File: CSharp/05. Longest Palindromic Substring (Medium)/Solution0.cs
using System;


namespace LeetCode
{
	class Solution0
	{

		public string LongestPalindrome(string str)
		{
			if (String.IsNullOrEmpty(str))
				return String.Empty;

			int n = str.Length;
			int longestStart = 0;
			int longestLength = 1;

			for (int i = 0; i < n; i++) {
				for (int j = n - 1; j >= i + longestLength; j--) {
					if (IsPalindrome(str, i, j)) {
						longestStart = i;
						longestLength = j - i + 1;
						continue;
					}
				}
			}

			return str.Substring(longestStart, longestLength);
		}

		static bool IsPalindrome(string str, int first, int last)
		{
			do {
				if (str[first++] != str[last--])
					return false;
			}
			while (first < last);

			return true;
		}
	}
}

File: Java/05. Longest Palindromic Substring (Medium)/src/LeetCode/LongestPalindromicSubstring/Solution0.java
/**
 * 
 */
package LeetCode.LongestPalindromicSubstring;

/**
 * @author Roman
 *
 */
public class Solution0
{
    public String longestPalindrome(String str)
    {
		int n = str.length();
		if (n == 0)
			return str;


		int longestStart = 0;
		int longestLength = 1;

		for (int i = 0; i < n; i++) {
			for (int j = n-1; j >= i + longestLength; j--) {
				if (isPalindrome(str, i, j)) {
					longestStart = i;
					longestLength = j - i + 1;
					continue;
				}
			}
		}

		return str.substring(longestStart, longestStart+longestLength);
    }

	static boolean isPalindrome(String str, int first, int last)
	{
		do {
			if (str.charAt(first++) != str.charAt(last--))
				return false;
		}
		while (first < last);

		return true;
	}
}
